src/main/java/com/find/doongji/like/service/BasicLikeService.java

```diff
@@ -0,0 +1,43 @@
+package com.find.doongji.like.service;
+
+import com.find.doongji.like.repository.LikeRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.nio.file.AccessDeniedException;
+
+@Service
+@RequiredArgsConstructor
+public class BasicLikeService implements LikeService {
+
+    private final LikeRepository likeRepository;
+
+    @Override
+    @Transactional
+    public void toggleLike(String aptSeq) throws AccessDeniedException {
+        if (!isLoggedIn()) {
+            throw new AccessDeniedException("You must be logged in to like.");
+        }
+
+        String username = SecurityContextHolder.getContext().getAuthentication().getName();
+        likeRepository.toggleLike(username, aptSeq);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public int viewLike(String aptSeq) {
+        if (!isLoggedIn()) {
+            return 0;
+        }
+
+        String username = SecurityContextHolder.getContext().getAuthentication().getName();
+        return likeRepository.selectLike(username, aptSeq);
+    }
```

Comment:
`SecurityContextHolder.getContext().getAuthentication().getName()` returns anonymousUser value instead of indicating lack of login status. How can I handle this?